### ISSUE TITLE
ArrayIndexOutofBoundsException when read directory from zipInputStream

### ISSUE THREAD

**Issue by IvanLi123** [srikanth-lingala/zip4j#33]
Hi, Author. I found out this bug that when I tried to read encrypted zip files from ZipInputStream, if it meet some directories, it would throw this exception:
java.lang.ArrayIndexOutOfBoundsException: 11
	at net.lingala.zip4j.util.RawIO.readShortLittleEndian(RawIO.java:109)
	at net.lingala.zip4j.headers.HeaderReader.parseExtraDataRecords(HeaderReader.java:309)
	at net.lingala.zip4j.headers.HeaderReader.readExtraDataRecords(HeaderReader.java:297)
	at net.lingala.zip4j.headers.HeaderReader.readExtraDataRecords(HeaderReader.java:264)
	at net.lingala.zip4j.headers.HeaderReader.readLocalFileHeader(HeaderReader.java:553)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:66)
	at net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry(ZipInputStream.java:62)

This error only happens on reading directories from ZipInputStream. 
Please help me fix this bug. I am in urgent need. 

Thank you in advance!

**Comment by IvanLi123** [comment]
[directory.zip](https://github.com/srikanth-lingala/zip4j/files/3449339/directory.zip)
Zip file attached.

**Comment by srikanth-lingala** [comment]
Can you please give me the password for the zip file and also the code you used? I have not been able to reproduce this issue with some sample zip files which have directories in it. For me ZipInputStream works fine.

**Comment by IvanLi123** [comment]
> Can you please give me the password for the zip file and also the code you used? I have not been able to reproduce this issue with some sample zip files which have directories in it. For me ZipInputStream works fine.

The password is password.
The code I used is:
```
   File file = new File("/home/yifan.li/Documents/directory.zip");
        FileInputStream fileInputStream = new FileInputStream(file);
        ZipInputStream zipInputStream = new ZipInputStream(fileInputStream, "password".toCharArray());
        LocalFileHeader localFileHeader;
        int readLen = 0;
        byte[] buffer = new byte[1024];
        while ((localFileHeader = zipInputStream.getNextEntry()) != null) {
            while ((readLen = zipInputStream.read(buffer)) != -1) {

            }
        }
```
Did I use it in a wrong way? 
Another issue I found is when I tried to read the password protected zip files on hdfs, I need to wrap the FSDataInputStream with BufferedInputStream or BytesInputStream. Otherwise it won't read anything. I know this API is only for the local files. Perhaps the problem is on the hdfs side.
Anyway, thank you for you reply and help.

**Comment by IvanLi123** [comment]
Hi, Srikanth, I am using Ubuntu. Before I just right clicked on the folder and clicked "compress" using this GUI to compress the folder. But when I tried to create the zip file from command line like "zip -re filename.zip folderpath password", it wouldn't throw this ArrayIndexOutofBoundsException.

That's weird.

**Comment by srikanth-lingala** [comment]
Apparently, Ubuntu is creating some corrupt extra data headers for some of the entries in the zip file. These headers are some additional information and the zip file can still be extracted without them. I have pushed a fix to ignore such headers. I will include this fix in the next release

**Comment by IvanLi123** [comment]
> Apparently, Ubuntu is creating some corrupt extra data headers for some of the entries in the zip file. These headers are some additional information and the zip file can still be extracted without them. I have pushed a fix to ignore such headers. I will include this fix in the next release

Thank you!

**Comment by srikanth-lingala** [comment]
Fixed in v2.1.2
